### ISSUE TITLE
Game crashes when interacting with Fruit

### ISSUE THREAD

**Issue by HawkSteak** [DynamicTreesTeam/DynamicTrees-PHC2#3]
Currently running into an issue on our server where interacting with fruit from PHC2 will cause that player to crash. This also extends to interactions with WAWLA/HWYLA, so looking at the fruit will also crash the player. Their crash log is attached. I believe this is related to the following errors that get thrown when the server first boots: [server errors.txt](https://github.com/DynamicTreesTeam/DynamicTrees-PHC2/files/11498457/server.errors.txt)

[crash-2023-05-06_00.57.37-fml.txt](https://github.com/DynamicTreesTeam/DynamicTrees-PHC2/files/11498416/crash-2023-05-06_00.57.37-fml.txt)

**Comment by HawkSteak** [comment]
Updating this to add that it looks like the names for the fruit are wrong within the code. It's currently looking for entities as `pamhc2trees:[fruitname]item` (ex: `pamhc2trees:almonditem`), but looking at the entity in singleplayer identifies it as `pamhc2trees:pam[fruitname]` (ex: `pamhc2trees:pamalmond`).

Even after attempting to tweak this though, the server still fails to find the registered blocks.

**Comment by AzoSBear** [comment]
I also saw that it fails to find the registry name for the harvest craft fruits in the latestlog file when booting up minecraft
[bootup log.txt](https://github.com/DynamicTreesTeam/DynamicTrees-PHC2/files/11523620/bootup.log.txt)

**Comment by supermassimo** [comment]
I am being unable to reproduce these bugs. Are you sure you are using the latest versions of all mods?
The crash seems to be related to some issues with the config files, not with the fruit.

**Comment by supermassimo** [comment]
The crash is a dup of https://github.com/DynamicTreesTeam/DynamicTrees-PHC2/issues/2 but I am leaving this open as the other registration issue is unrelated

**Comment by HawkSteak** [comment]
> 

Just did a fresh server install to test and still receiving the same errors. I removed all mods but the following, which are all up-to-date (version number listed as well for good measure). Configs were not touched aside from changing the world name and setting world generator to Biomes O Plenty.

Forge (36.2.39)
Biomes O Plenty (13.1.0.488)
Dynamic Trees (0.10.5)
Dynamic Trees Plus (0.1.0)
Dynamic Trees BOP (2.0.9)
Dynamic Trees PHC2 (1.0.0)
Pam HC2 Food Core (1.0.2)
Pam HC2 Food Extended (1.0.4)
Pam HC2 Crops (1.0.2)
Pam HC2 Trees (1.0.1)

[server.errors.txt](https://github.com/DynamicTreesTeam/DynamicTrees-PHC2/files/11524961/server.errors.txt)

**Comment by HawkSteak** [comment]
Also able to replicate it with only Dynamic Trees PHC2 and its dependencies (DT, PHC2, PHC2trees) installed. Client side is fine unless joining the server without loading singleplayer beforehand.

**Comment by supermassimo** [comment]
Ohh I see, so to reproduce join a server without joining singleplayer first

**Comment by HawkSteak** [comment]
> Ohh I see, so to reproduce join a server without joining singleplayer first

Yup! The server throws the errors attached to the last post on boot, but it only impacts/crashes the client if they go to multiplayer first.

**Comment by Squidvad3r** [comment]
A friend of mine is getting the same issue. Whenever she looks at fruit, specifically from PHC2:Dynamic trees, she crashes. 
The same happens to me if I join her server.

However, if you load a single-player world and look at a fruit tree (idk if the 'looking' part is needed, knock on wood), then join a server, you won't crash from looking at a random pear/apricot/gooseberry.

So, yeah, launch single-player, look at some fruit, then quit and you can play on multiplayer. It's a 'Janky fix' in her words, but idk what she's complaining about.

Hope this helps :)

**Comment by supermassimo** [comment]
Yeah this is unfortunately a side effect of how PHC2 registers it's items in a very non-standard way. We are still looking into a fix from our side

**Comment by Cryptillian** [comment]
Just wanted to bump this that my friend is having this issue on our LAN server as well when he looks directly at fruit on a tree (specifically he was looking at gooseberries).

**Comment by Likeabozz123** [comment]
same issue still

**Comment by Foulest** [comment]
Bump.

**Comment by Foulest** [comment]
Fixed in [my fork of DynamicTrees](https://github.com/Foulest/DynamicTrees) for 1.16.5.

**Comment by supermassimo** [comment]
@Foulest what was the change needed to fix this? Tried to look for it in your fork's commits but they are too large to efficiently find all the changes

**Comment by Foulest** [comment]
> @Foulest what was the change needed to fix this? Tried to look for it in your fork's commits but they are too large to efficiently find all the changes

https://github.com/Foulest/DynamicTrees/blob/8a16d73c49b4c7a5256623f98d60f2a539c3c3af/src/main/java/com/ferreusveritas/dynamictrees/systems/fruit/Fruit.java#L179

**Comment by Andreykoder** [comment]
Не открывается ссылка
